src/main/java/ch/njol/skript/expressions/ExprEntitySnapshot.java

```diff
@@ -0,0 +1,49 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntitySnapshot;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Entity Snapshot")
+@Description({
+	"Returns an entity snapshot of the provided entities, which is a snapshot of all the data of the provided entities.",
+	"This data includes the values of all the entities' attributes at the time of this expression being called.",
+	"This expression can only be used to copy the full attributes of an entity. Individual attributes cannot be modified or retrieved."
+})
+@Examples({
+	"spawn a pig at location(0, 0, 0):",
+		"\tset the max health of entity to 20",
+		"\tset the health of entity to 20",
+		"\tset {_snapshot} to the entity snapshot of entity",
+		"\tclear entity",
+	"spawn {_snapshot} at location(0, 0, 0)"
```

Comment:
I feel like there's a better example out there, but I can't think of one right now.